### PR TITLE
Implement History into Authorizations and State Calculation

### PR DESCRIPTION
**Pull request by florianebmeier**

### CODE REVIEW

File: Authorization/History/Create.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	reason: Error.Code
+	error: gracely.Error
+}
+interface Success extends Base {
+	verification?: "verified" | "rejected" | "unavailable"
+	status: "success"
+}
+interface Pending extends Base {
+	status: "pending"
+	rule: string
```
Comment (Magnus93) [inline Authorization/History/Create.ts:35]: Change `rule: string[]`

File: Authorization/History/Refund.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		return (
+			typeof value == "object" &&
+			value.status == "fail" &&
+			Error.Code.is(value.reason) &&
+			gracely.Error.is(value.error) &&
+			isBase(value)
+		)
+	}
+	export function isSuccess(value: any | Success): value is Success {
+		return typeof value == "object" && value.status == "success" && typeof value.status == "string" && isBase(value)
```
Comment (Magnus93) [inline Authorization/History/Refund.ts:49]: `value.status == "success"` and `typeof value.status == "string"` is redundant.
Remove: `typeof value.status == "string"`

File: Authorization/index.ts [hunk 3]
```diff
@@ -160,7 +165,7 @@ export namespace Authorization {
 		result += "card expires,"
 		result += "descriptor,"
 		result += "recurring,"
-		result += "history,"
+		result += "change,"
```
Comment (Magnus93) [inline Authorization/index.ts:168]: Should new history be in toCsv function?

Comment (florianebmeier) [reply Authorization/index.ts:168]: No

File: Authorization/History/Create.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	card?: Card
+	descriptor?: string
+	recurring?: Recurring
+	verification?: "verified" | "rejected" | "unavailable"
+	rule?: string[]
+	reason: Error.Code
+	error: gracely.Error
+}
+interface Success extends Base {
+	verification?: "verified" | "rejected" | "unavailable"
```
Comment (Magnus93) [inline Authorization/History/Create.ts:30]: What is the difference bewteen `verification = "unavailable"` vs `verification = undefined` ?

Comment (Magnus93) [inline Authorization/History/Create.ts:30]: What is the difference between `verification = "unavailable"` vs `verification = undefined` ?

Comment (florianebmeier) [reply Authorization/History/Create.ts:30]: One is that the service to verify your identity is unavailable, the other is that there is no verification, i.e. verification has not been tried at all

**Review by Magnus93 — APPROVED** [review]

**Review by nilssonemma — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**